crates/swc_ecma_minifier/tests/fixture/next/syncfusion/933-e9f9a6bf671b96fc/output.js

```diff
@@ -7214,33 +7214,34 @@
             }
             function calculatePosition(currentElement, positionX, positionY, parentElement, targetValues) {
                 return (popupRect = void 0, popupRect = targetValues, fixedParent = !!parentElement, currentElement) ? (positionX || (positionX = 'left'), positionY || (positionY = 'top'), parentDocument = currentElement.ownerDocument, element = currentElement, function(posX, posY, pos) {
+                    var value, value1, value2, value3, value4, value5, value6, value7, value8;
                     switch(elementRect = element.getBoundingClientRect(), posY + posX){
                         case 'topcenter':
-                            setPosx(getElementHCenter(), pos), setPosy(getElementTop(), pos);
+                            setPosx(getElementHCenter(), pos), value = getElementTop(), pos.top = value;
                             break;
                         case 'topright':
-                            setPosx(getElementRight(), pos), setPosy(getElementTop(), pos);
+                            setPosx(getElementRight(), pos), value1 = getElementTop(), pos.top = value1;
                             break;
                         case 'centercenter':
-                            setPosx(getElementHCenter(), pos), setPosy(getElementVCenter(), pos);
+                            setPosx(getElementHCenter(), pos), value2 = getElementVCenter(), pos.top = value2;
                             break;
                         case 'centerright':
-                            setPosx(getElementRight(), pos), setPosy(getElementVCenter(), pos);
+                            setPosx(getElementRight(), pos), value3 = getElementVCenter(), pos.top = value3;
                             break;
                         case 'centerleft':
-                            setPosx(getElementLeft(), pos), setPosy(getElementVCenter(), pos);
+                            setPosx(getElementLeft(), pos), value4 = getElementVCenter(), pos.top = value4;
                             break;
                         case 'bottomcenter':
-                            setPosx(getElementHCenter(), pos), setPosy(getElementBottom(), pos);
+                            setPosx(getElementHCenter(), pos), value5 = getElementBottom(), pos.top = value5;
                             break;
                         case 'bottomright':
-                            setPosx(getElementRight(), pos), setPosy(getElementBottom(), pos);
+                            setPosx(getElementRight(), pos), value6 = getElementBottom(), pos.top = value6;
                             break;
                         case 'bottomleft':
-                            setPosx(getElementLeft(), pos), setPosy(getElementBottom(), pos);
+                            setPosx(getElementLeft(), pos), value7 = getElementBottom(), pos.top = value7;
```

Comment:
This looks like a regresion. Are we considering the size of the member property correctly?

Reply:
We count `pos.top = value` as size 7, as both `pos` and `value` can be mangled, and function call cost is `3 + param_count * 2` equals 7, so it got inlined. This regression is caused by `value` cannot be inlined, as `pos.top` may have side effect. However we cannot know if an argument has side effect at the time of inlining since we only know it's declaration.